shell/common/shell.cc

```diff
@@ -1332,6 +1332,18 @@ void Shell::OnEngineHandlePlatformMessage(
   }
 }
 
+void Shell::OnEngineChannelListenedTo(const std::string& name, bool listening) {
+  FML_DCHECK(is_set_up_);
+  FML_DCHECK(task_runners_.GetUITaskRunner()->RunsTasksOnCurrentThread());
+
+  task_runners_.GetPlatformTaskRunner()->PostTask(
```

Comment:
I'm a bit worried about posting a task to the platform thread that captures a `const std::string &` that was created by Dart FFI.

`PlatformConfigurationNativeApi::SendPlatformMessage` has a similar problem, and I see that it copies the channel's `std::string` ([here](https://github.com/flutter/engine/blob/711ebac54d095717f3448cda5cf73fd99153e4a1/lib/ui/window/platform_configuration.cc#L376) and [here](https://github.com/flutter/engine/blob/83fdadaf12f00ef0ca3aa9401918a2486fcb89a2/lib/ui/window/platform_message.cc#L11))

Should we also copy the channel's `std::string`? Also, should we `std::move` this copied string down instead of passing it by reference?